compiler/fir/resolve/src/org/jetbrains/kotlin/fir/declarations/InlineClassesUtils.kt

```diff
@@ -11,21 +11,30 @@ import org.jetbrains.kotlin.fir.resolve.substitution.createTypeSubstitutorByType
 import org.jetbrains.kotlin.fir.resolve.toSymbol
 import org.jetbrains.kotlin.fir.resolve.transformers.ensureResolved
 import org.jetbrains.kotlin.fir.symbols.impl.FirRegularClassSymbol
-import org.jetbrains.kotlin.fir.types.*
+import org.jetbrains.kotlin.fir.types.ConeKotlinType
+import org.jetbrains.kotlin.fir.types.ConeLookupTagBasedType
+import org.jetbrains.kotlin.fir.types.ConeTypeContext
+import org.jetbrains.kotlin.fir.types.coneType
 import org.jetbrains.kotlin.types.model.typeConstructor
 
 fun ConeKotlinType.substitutedUnderlyingTypeForInlineClass(session: FirSession, context: ConeTypeContext): ConeKotlinType? {
-    val symbol = (this.fullyExpandedType(session) as? ConeLookupTagBasedType)
-        ?.lookupTag
-        ?.toSymbol(session) as? FirRegularClassSymbol
-        ?: return null
-    symbol.ensureResolved(FirResolvePhase.STATUS, session)
-    val firClass = symbol.fir
-    if (!firClass.status.isInline) return null
+    val firClass = toFirRegularClass(session) ?: return null
+    val inline = firClass.status.isInline
+    if (!inline) return null
     val constructor = firClass.declarations.singleOrNull { it is FirConstructor && it.isPrimary } as FirConstructor? ?: return null
     val valueParameter = constructor.valueParameters.singleOrNull() ?: return null
     val unsubstitutedType = valueParameter.returnTypeRef.coneType
 
     val substitutor = createTypeSubstitutorByTypeConstructor(mapOf(this.typeConstructor(context) to this), context)
     return substitutor.substituteOrNull(unsubstitutedType)
 }
+
+fun ConeKotlinType.toFirRegularClass(session: FirSession): FirRegularClass? {
```

Comment:
Maybe in a separate commit, you can clean up usage of this one: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/FirHelpers.kt#L115

or use that one in checkers. But, we may need to use `fullyExpandedType` there as well.

Reply:
OK, let me update the one in FirHelpers.kt and use it. I will revert this change here. I can't make this call the helper in FirHelpers since this is in the `resolve` module and cannot depend on the `checkers` module.